tools/lambda-promtail/lambda-promtail/promtail.go

```diff
@@ -129,7 +129,7 @@ func (b *batch) flushBatch(ctx context.Context) error {
 	}
 
 	b.streams = make(map[string]*logproto.Stream)
-
+	b.size = 0
```

Comment:
what is this for? fixing one of the bugs you're referring to I assume?

Reply:
> what is this for? fixing one of the bugs you're referring to I assume?

Yes because batch.size is incremented as we scan the file and after each added line we compare batch.size and (max)batchSize to know if we need to flush the batch to the target (loki/promtail). 

It needs to be reset (to 0) after batch.size gets bigger than batchSize otherwise we end up sending a request to the target to flush a single line for each new line (so it becomes super slow because of the thousands of round-trip calls)

I think I will update this PR to provide a proper reset() method to make it clearer

Reply:
a `resetBatch` function sounds like a good idea :+1:

Reply:
Done in my latest commit 👍